x/vesting/keeper/integration_test.go

```diff
@@ -0,0 +1,515 @@
+	validators := app.StakingKeeper.GetValidators(suite.ctx, 2)
+	// Set a bonded validator that takes part in consensus
+	if validators[0].Status == stakingtypes.Bonded {
+		suite.validator = validators[0]
+	} else {
+		suite.validator = validators[1]
+	}
+	return nil
+}
+
+var _ = Describe("Monthly Vesting Account", Ordered, func() {
+	const (
+		cliffDays = 30
+		months    = 3
+	)
+
+	var (
+		s *VestingTestSuite
+
+		vaPrivKey cryptotypes.PrivKey // private key of vesting account
+		va        sdk.AccAddress      // vesting account
+		funder    sdk.AccAddress      // funder account who initiates monthly vesting
+		user      sdk.AccAddress
+
+		initialVesting sdk.Coins // initial vesting coins
+		extraCoins     sdk.Coins // additional funds to pay gas fees for tx
+		mva            *types.MonthlyVestingAccount
+
+		unvested sdk.Coins
+		vested   sdk.Coins
+	)
+
+	BeforeEach(func() {
+		var err error
+
+		s = new(VestingTestSuite)
+		err = s.SetupTest()
+		Expect(err).To(BeNil())
+
+		now := s.ctx.BlockTime()
+
+		var from common.Address
+		from, vaPrivKey = tests.RandomEthAddressWithPrivateKey()
+		va = sdk.AccAddress(from.Bytes())
+		user = tests.RandomAccAddress()
+		funder = tests.RandomAccAddress()
+
+		amount := math.NewInt(1e17).Mul(math.NewInt(months))
+		initialVesting = sdk.NewCoins(sdk.NewCoin(utils.BaseDenom, amount))
+
+		// Fund coins to funder to create monthly vesting
+		err = testutil.FundAccount(s.ctx, s.app.BankKeeper, funder, initialVesting)
+		Expect(err).To(BeNil())
+		err = s.Commit()
+
+		// Create monthly vesting
+		resp, err := s.msgServer.HandleCreateMonthlyVestingAccount(s.goCtx, &types.MsgCreateMonthlyVestingAccount{
+			FromAddress: funder.String(),
+			ToAddress:   va.String(),
+			CliffDays:   cliffDays, // 30 days
+			Months:      months,    // 3 months
+			Amount:      initialVesting,
+		})
+		Expect(resp).To(Equal(&types.MsgCreateMonthlyVestingAccountResponse{}))
+		err = s.Commit()
+		Expect(err).To(BeNil())
+
+		// Query monthly vesting account
+		mva, err = s.querier.GetMonthlyVestingAccount(s.ctx, va)
+		Expect(err).To(BeNil())
+		Expect(mva).ToNot(BeNil())
+
+		//baseAcc := authtypes.NewBaseAccountWithAddress(va)
+		//
+		//// Cliff days = 30, Months = 12
+		//mva = types.NewMonthlyVestingAccount(baseAcc, initialVesting, now.Unix(), cliffDays, months)
+		//acc := s.app.AccountKeeper.NewAccount(s.ctx, mva)
+		//s.app.AccountKeeper.SetAccount(s.ctx, acc)
+		//
+		//err = testutil.FundAccount(s.ctx, s.app.BankKeeper, mva.GetAddress(), initialVesting)
+		//Expect(err).To(BeNil())
+		//err = s.Commit()
+		//Expect(err).To(BeNil())
+
+		// Fund again as spendable coins for gas consuming
+		extraCoins = sdk.NewCoins(sdk.NewCoin(utils.BaseDenom, math.NewInt(1e6)))
+		err = testutil.FundAccount(s.ctx, s.app.BankKeeper, mva.GetAddress(), extraCoins)
+		Expect(err).To(BeNil())
+		err = s.Commit()
+		Expect(err).To(BeNil())
+
+		// Check if all the tokens are unvested at beginning
+		unvested := mva.GetVestingCoins(now)
```

Comment:
What is the difference between `vested` and `unvested` vars. Looks like they are equal